Core-SDK/Assets/MoEngage/Scripts/MoEUtils.cs

```diff
@@ -18,210 +18,232 @@
-    {
-			Dictionary<string, object> payloadDictionary = MoEMiniJSON.Json.Deserialize(payload) as Dictionary<string, object>;
-			
-			return new PushToken(
-				payloadDictionary[MoEConstants.PARAM_PLATFORM] as string,
-				payloadDictionary[MoEConstants.PARAM_PUSH_TOKEN] as string,
-				(PushService)Enum.Parse(typeof(PushService), payloadDictionary[MoEConstants.PARAM_PUSH_SERVICE] as string)
-				);
-    }
+                campaign.selfHandled = selfHandled;
+            }
+
+            return campaign;
+        }
+
+        public static string GetSelfHandledPayload(InAppCampaign inAppCampaign, string type)
+        {
+            var selfHandledDictionary = new Dictionary<string, object>()
+            {
+                { MoEConstants.ARGUMENT_PAYLOAD,  inAppCampaign.selfHandled.payload},
+                { MoEConstants.ARGUMENT_DISMISS_INTERVAL, inAppCampaign.selfHandled.dismissInterval },
+                { MoEConstants.ARGUMENT_IS_CANCELLABLE, inAppCampaign.selfHandled.isCancellable}
+            };
+            var inAppCampaignDictionary = new Dictionary<string, object>()
+            {
+                { MoEConstants.ARGUMENT_CAMPAIGN_ID, inAppCampaign.campaignId },
+                { MoEConstants.ARGUMENT_CAMPAIGN_NAME, inAppCampaign.campaignName},
+                { MoEConstants.ARGUMENT_SELF_HANDLED, selfHandledDictionary},
+                { MoEConstants.ARGUMENT_CAMPAIGN_CONTEXT,inAppCampaign.campaignContext},
+                { MoEConstants.ARGUMENT_TYPE, type}
+            };
+            return Json.Serialize(inAppCampaignDictionary);
+        }
+
+        public static string GetOptOutTrackingPayload(string type, bool shouldOptOut)
+        {
+            var optOutTrackingDictionary = new Dictionary<string, object>()
+            {
+                { MoEConstants.ARGUMENT_TYPE, type },
+                { MoEConstants.PARAM_STATE, shouldOptOut}
+            };
+
+            return Json.Serialize(optOutTrackingDictionary);
+        }
+
+        public static string GetSdkStatePayload(bool isSdkEnabled)
+        {
+            var sdkStatusDictionary = new Dictionary<string, object>()
+            {
+                { MoEConstants.FEATURE_STATUS_IS_SDK_ENABLED, isSdkEnabled},
+            };
+
+            return Json.Serialize(sdkStatusDictionary);
+        }
+
+        public static PushToken GetPushTokenFromPayload(string payload)
+        {
+            Dictionary<string, object> payloadDictionary = MoEMiniJSON.Json.Deserialize(payload) as Dictionary<string, object>;
+
+            return new PushToken(
+                payloadDictionary[MoEConstants.PARAM_PLATFORM] as string,
+                payloadDictionary[MoEConstants.PARAM_PUSH_TOKEN] as string,
+                (PushService)Enum.Parse(typeof(PushService), payloadDictionary[MoEConstants.PARAM_PUSH_SERVICE] as string)
+                );
+        }
+
+        public static string GetAndroidIdTrackingStatus(bool isEnabled)
```

Comment:
@umangmoe APIs `GetAndroidIdTrackingStatus` and `GetAdIdTrackingStatus` are the newly added ones, rest APIs remain umchanged